forastero/bench.py

```diff
@@ -376,3 +400,18 @@ def _inner(method: Callable) -> Callable:
             return method
 
         return _inner
+
+# Start profiling when it is enabled in the parameters file
+if (outfile := BaseBench.PARSED_PARAMS.get("profiling")):
+    import atexit, yappi
+    logging.warning("Profiling has been enabled")
+    yappi.set_clock_type("cpu")
+    yappi.start()
+    # Register a teardown method to stop profiling when Python exits
+    def _end_profile():
+        yappi.stop()
+        logging.warning("Profiling summary:")
+        logging.info(yappi.get_func_stats().print_all())
+        logging.warning(f"Profile data written to {outfile}")
```

Comment:
Warning doesn't feel quite right

Reply:
(but in absence of something better I'm not particularly bothered)

Reply:
Changed to `logging.info` - I've left the "Profiling enabled" message as a warning as it absolutely kills performance when enabled